### DISCUSSION TITLE
New Scalable Keyboard

### DISCUSSION THREAD

**Discussion by sofia-bg** [UNC-Project-Open-AAC/OS-DPI#257] [Help]
Here is a new iteration of the scalable keyboard design based on a different keyboard. I have it mostly working, for some reason I cannot get delete last work to work. @gbishop I am wondering if you have ideas about how to automatically save recent messages to that grid just below the display? 

[Grid3_10_15_24_v2.zip](https://github.com/user-attachments/files/17385394/Grid3_10_15_24_v2.zip)

**Comment by gbishop** [comment]
Delete word was not working because you had #Display instead of $Display. I
should make a diagnostic for that...

You could use the Notes machinery to populate that grid. See
https://github.com/UNC-Project-Open-AAC/OS-DPI/tree/main/Designs%20In%20Progress/Notes

Look at the Actions for save and delete.

[image: image.png]

And at the Notes grid in the layout.

[image: image.png]

I'll be pleased to meet with you if it'll help.

gb

**Reply by sofia-bg** [reply]
Thank you! I was comparing line by line to a layout where it was working and I just didn't catch that difference. 

I'll give the notes a try over the next few days and reach out if I get stuck.

**Reply by sofia-bg** [reply]
I have it working emulating what you did with notes. When you press the display to speak it automatically saves what you said to the empty row of cells below the display. However, it only saves to the first button, replacing that button each time you speak. How can I set it up so that on the next click it saves to the next button? 

[Grid3_Patrick_10_15_24_v2.zip](https://github.com/user-attachments/files/17578830/Grid3_Patrick_10_15_24_v2.zip)

**Comment by gbishop** [comment]
A letter on my Notes demo got deleted. In the action for Clear the origin should be "controls" not "control". 

That said, the issue in your design is the $id state. If it has a value, then it overwrites the note with that #ID. If the $id is empty then it creates a new Note. By keeping the value of $id you can edit a Note. By clearing the value you can make a new one. 

So you need to add an update to one of your actions to clear the $id.

**Comment by sofia-bg** [comment]
@lageist here is the most current version 
[Grid3_Patrick_11_4_24_v3.zip](https://github.com/user-attachments/files/18085869/Grid3_Patrick_11_4_24_v3.zip)

**Reply by sofia-bg** [reply]
I lied - the newest one is here: 
[Grid3_Patrick_12.10.24.zip](https://github.com/user-attachments/files/18086076/Grid3_Patrick_12.10.24.zip)

**Reply by lageist** [reply]
@gbishop  can you please help (or remind me how) to make the above simulation available as a direct link that opens in the platform? Sharing with our development partner. Thanks.

**Comment by gbishop** [comment]
https://unc-project-open-aac.github.io/OS-DPI/?fetch=https://github.com/user-attachments/files/18086076/Grid3_Patrick_12.10.24.zip

Seems to work for me. The pattern is

https://unc-project-open-aac.github.io/OS-DPI/?fetch=

followed by the URL of the design.

**Reply by lageist** [reply]
@gbishop thanks!
